tests/evaluate/test_evaluation.py

```diff
@@ -1365,3 +1365,97 @@ def test_evaluate_with_loaded_pyfunc_model():
+    X = X[::5]
+    y = y[::5]
+    with mlflow.start_run() as run:
+        mlflow.evaluate(
+            data=X.assign(y=y, model_output=y),
+            targets="y",
+            predictions="model_output",
+            model_type="regressor",
+        )
+
+    run = mlflow.get_run(run.info.run_id)
+    assert "mean_absolute_error" in run.data.metrics
+    assert "mean_squared_error" in run.data.metrics
+    assert "root_mean_squared_error" in run.data.metrics
+
+
+def test_evaluate_with_static_spark_dataset_unsupported():
+    data = sklearn.datasets.load_diabetes()
+    spark = SparkSession.builder.master("local[*]").getOrCreate()
+    rows = [
+        (Vectors.dense(features), float(label), float(label))
+        for features, label in zip(data.data, data.target)
+    ]
+    spark_dataframe = spark.createDataFrame(
+        spark.sparkContext.parallelize(rows, 1), ["features", "label", "model_output"]
+    )
+    with mlflow.start_run():
+        with pytest.raises(
+            MlflowException, match="The data must be a pandas dataframe when model=None."
+        ):
+            mlflow.evaluate(
+                data=spark_dataframe,
+                targets="label",
+                predictions="model_output",
+                model_type="regressor",
+            )
+
+
+def test_evaluate_with_static_dataset_error_handling():
+    X, y = sklearn.datasets.load_diabetes(return_X_y=True, as_frame=True)
+    X = X[::5]
+    y = y[::5]
+    with mlflow.start_run():
+        with pytest.raises(
+            MlflowException, match="The predictions argument must be specified when model=None."
+        ):
+            mlflow.evaluate(
+                data=X.assign(y=y, model_output=y),
+                targets="y",
+                model_type="regressor",
+            )
+
+        with pytest.raises(
+            MlflowException, match="The data must be a pandas dataframe when model=None."
+        ):
+            mlflow.evaluate(
+                data=X.assign(y=y, model_output=y).to_numpy(),
+                targets="y",
+                predictions="model_output",
+                model_type="regressor",
+            )
+
+        with pytest.raises(
+            MlflowException,
+            match="The predictions argument cannot be specified when model is specified.",
+        ):
+            mlflow.evaluate(
+                model="models:/test",
+                data=X.assign(y=y, model_output=y).to_numpy(),
+                targets="y",
+                predictions="model_output",
+                model_type="regressor",
+            )
+
+        with pytest.raises(MlflowException, match="The data argument cannot be None."):
+            mlflow.evaluate(
+                data=None,
+                targets="y",
+                model_type="regressor",
+            )
+
+        with pytest.raises(
+            MlflowException, match="The predictions column prediction is not found in data."
```

Comment:
```suggestion
            MlflowException, match="The specified predictions column 'prediction' is not found in the specified data."
```

Can we make similar changes to other error messages, i.e. use single quotes for user-specified values?

Reply:
Updated